### ISSUE TITLE
Next Release Checklist

### ISSUE THREAD

**Issue by wez** [wezterm/wezterm#6341]
**BEFORE YOU COMMENT: please avoid duplicating information from other comments to avoid overwhelming this one.**
Prefer to use a thumbs up instead of a me-too comment.

We're overdue for making a new release.
Here's what's on my mind as release gating issues, and I need your help to check these off.
So if you are interested in having a new stable tag, please help by running through the list below and giving some feedback on the state of these things for your system:

* [ ] Verify that Wayland support isn't an obvious regression since the last stable release.
     *  Wayland support was essentially rewritten by the community, and since I don't use wayland, I don't have sufficient context now to understand where it has problems either in the code or due to compatibility.
     * If you are a Wayland user, and you are using a wezterm nightly build from October 2024 or later and having a good experience, please comment below with the name of your compositor and DE to say that it is as good as or better than the prior stable release.
     * If there is already a comment describing your environment, instead of posting a duplicate, please give it a thumbs-up reaction to "vote" for it if it is working well for you.
     * If it isn't working working as well or better than the prior stable release, please reference an issue that describes the problem and comment below.

* [ ] Verify that the default configuration doesn't crash out.
     * Run `wezterm -n` on your system to launch it without your config.  Does it successfully launch and display text with no artifacts or blocks for glyphs?
     * If not:
         *  try changing the [front_end](https://wezfurlong.org/wezterm/config/lua/config/front_end.html) setting to see if that is better:
        * `wezterm -n --config "front_end='WebGpu'"`
        * Please find/file an issue reporting the outcome of both the default and webgpu, along with information about the environment. **If the behavior is a regression since the prior stable release**, reference that issue here.
 
* [ ] Confirm that we're building for the current batch of OS/distros
    * https://github.com/wez/wezterm/pull/6156
    * https://github.com/wez/wezterm/pull/6200
    * Others? Please comment here 

* Other misc issues:
   * https://github.com/wez/wezterm/pull/6186

**Comment by rootiest** [comment]
###  Specs
KDE Plasma Version: 6.2.2
KDE Frameworks Version: 6.7.0
Qt Version: 6.8.0
Kernel Version: 6.11.5-zen1-1-zen (64-bit)
Graphics Platform: Wayland

Processors: 16 x AMD Ryzen 9 4900HS with Radeon Graphics
Memory: 22.9 GiB of RAM
Graphics Processor: AMD Radeon Graphics (+ Nvidia RTX 2060)
Manufacturer: ASUSTeK COMPUTER INC.
Product Name: ROG Zephyrus G14 GA401IV_GA401IV
System Version: 1.0

WezTerm version: wezterm 20241015-083151-9ddca7bd

### Experience

By and large everything works flawlessly for me.
With or without WebGpu, and with or without my config loaded.

The only issue I have experienced is if I maximize the WezTerm window on my external display with a 1440p resolution, then unplug that display and use my laptop's 1080p display, WezTerm will still try to maximize to the original resolution, even if I un-maximize and re-maximize it. It seems to be locked to that original size for maximized windows until I close and re-open the terminal window.

**Comment by rootiest** [comment]
KDE Plasma 6 (Wayland) working great for me.

**Comment by jankatins** [comment]
I'm on Gnome 46/Mutter with Windowing system Wayland (fedora 40 workstation, up to date until yesterday), using wezterm 20241027_002434_14ad1937.

Stuff which I "notice" (I seem to remember that it worked sometime in the past or when I was still on macosx). This is all with `config.enable_wayland = false` or not having that config set at all:

- I don't have an additional title bar (which is fine-ish) and double click on the tab bar to maximize does not work (which is bad). I added that as a config in Gnome: https://github.com/wez/wezterm/issues/2831
- I don't have min/max/close buttons in the tab bar, which I kind of think where there earlier?
- I cannot move the window anymore (does that mean weztemr ist actually otherwise running NOT in wayland? https://github.com/wez/wezterm/issues/6296 
- I use a `gui-startup` event handler to `window:gui_window():maximize()` -> my window still comes up unmaximized -> Prob https://github.com/wez/wezterm/issues/6203
- I don't have any min/max/close buttons and need to use 'win+up' to toogle maximize (which I configured in gnome)
- If I configure `config.window_decorations = "INTEGRATED_BUTTONS"`, I get an additional (blackmenu bar with the min/max/close buttons and max and close work.  I can also move the window around when I start in that new menu bar. Still, double click on that menu bar to maximize does NOT work and the min button also does NOT work: https://github.com/wez/wezterm/issues/6086 (this might be a general error, as the issue was reported on x11)
- With `config.window_decorations = "INTEGRATED_BUTTONS"` The full screen window "overflows" at the bottom. https://github.com/wez/wezterm/issues/6318

If I run with `config.enable_wayland = false`, I have a (grey) additional menu bar with a single close button, I can move the window around, even when I was in fullscreen, it starts in full screen, and the window overflow problem is also gone.

Relevant Gnome config:

```shell
gsettings set org.gnome.desktop.wm.keybindings maximize "['<Super>Up']"
gsettings set org.gnome.desktop.wm.preferences action-double-click-titlebar 'toggle-maximize'
```

**Comment by F5T3** [comment]
wezterm-git (AUR) on Hyprland (v0.44.0-73-g3cec45d8) on Arch Linux (Kernerls: linux [6.11.5-arch1-1], linux-lts [6.6.58-1] and linux-zen [6.11.5.zen1-1])

**Comment by nekowinston** [comment]
There's a regression on [Sway 1.10](https://github.com/swaywm/sway/releases/tag/1.10).
Nightly works for me on Sway 1.9, but I'm guessing the new wlroots-0.18 in 1.10 broke parts of the WezTerm Wayland implementation.

|           | WezTerm stable   | WezTerm nightly ([f847bd28][nightly]) |
|-----------|------------------|---------------------------------------|
| Sway 1.9  | ❌ [Logs][log-1] | ✅ [Logs][log-2]                      |
| Sway 1.10 | ❌ [Logs][log-3] | ❌ [Logs][log-4]                      |

Tested on NixOS unstable, Linux 6.11.5, x86_64-linux.

All logs are collected via:
```console
WAYLAND_DEBUG=1 WEZTERM_LOG=window::os::wayland=trace cargo r --release --bin wezterm-gui -- -n --config "front_end='WebGpu'"
```

It's worth mentioning that nighly@1.9 initially renders blurry text, which goes away after scaling/moving/maximizing the window once. But overall, I had a pretty reliable experience on Wayland on Sway 1.9.

---

As for Sway 1.10:

I've tried to chase down the error with a debug build & via debugging via CodeLLDB, all I could gather from it, is that it crashes around [here](https://github.com/wez/wezterm/blob/f847bd284db6a38e7f5920421ba23e3aa688f8fd/window/src/os/wayland/connection.rs#L77-L120).
It seems to crash after a few iterations, so it makes debugging a bit frustrating; I'm happy to dive back in tho.

[nightly]: https://github.com/wez/wezterm/commit/f847bd284db6a38e7f5920421ba23e3aa688f8fd
[log-1]: https://gist.github.com/nekowinston/aed218bef0f15d9557979c886c25943a#file-log-stable-sway1-9-log
[log-2]: https://gist.github.com/nekowinston/aed218bef0f15d9557979c886c25943a#file-log-nightly-sway1-9-log
[log-3]: https://gist.github.com/nekowinston/aed218bef0f15d9557979c886c25943a#file-log-stable-sway1-10-log
[log-4]: https://gist.github.com/nekowinston/aed218bef0f15d9557979c886c25943a#file-log-nightly-sway1-10-log

**Comment by wez** [comment]
Thanks for commenting folks.  From a triage perspective, what I care about here in the context of this issue is:

* _Would releasing nightly now be worse than the prior stable release?_

I don't want this issue to an umbrella for current bugs: those can all continue to live in the issue tracker.

**So please review the comments you have made and if the issues are also present in the prior stable release, please edit your comments to elide that information from here.**

I want to use this issue as a point of triage for getting wezterm-regressions-in-behavior-vs-its-last-stable-release resolved, and if it just duplicates general bugs it is not helpful.

Thank you!

**Comment by RomanAverin** [comment]
wezterm 20241031_235112_0983ae90
Fedora 41 - Gnome 47.1 - Mutter 47.1 - Wayland

- From what I feel the night version is a bit more stable(less crash), but in terms of interface it is a clear regression compared to the last release version 20240203_110809_5046fc22. Window resizing doesn't work in the night version as part of the regression.

![Screenshot From 2024-11-02 12-44-50](https://github.com/user-attachments/assets/c260a597-c20a-4761-9b94-c7cc5c56e057)

![Screenshot From 2024-11-02 12-45-59](https://github.com/user-attachments/assets/e2ae9e73-8495-4b7f-a74c-9dd047380cb2)
```
config.animation_fps = 60
config.front_end = "WebGpu"
config.window_decorations = "INTEGRATED_BUTTONS"
```

**Comment by ateijelo** [comment]
Nightly fails to connect to an IPv6 link-local hostname that stable was connecting to successfully: https://github.com/wez/wezterm/issues/5934#issuecomment-2454899213

**Comment by Felixoid** [comment]
I am using Plasma 6 on arch linux, and have found so far two issues:

- An already mentioned https://github.com/wez/wezterm/issues/6203, and it looks a bit messy
- Annoying issue with moving wezterm window from one Wayland display to another https://github.com/wez/wezterm/issues/6374

**Comment by rlcurrall** [comment]
I used the nightly version on Windows 11 for about a week. I switched back to the sable release because my 1Password SSH integration wasn't working since the build wasn't signed in a way that Windows/1Password recognized.

During that I didn't notice any regressions in functionality or performance (granted my configuration is pretty light). The main benefit of switching to the nightly for me was getting proper undercurl support so linting showed up in NeoVim. I'm currently using Windows Terminal while waiting for the next tagged release so I can use NeoVim and have linting work for me, but I'm eagerly awaiting a new release.

**Comment by NikSneMC** [comment]
WezTerm Nightly works fine for me on Hyprland 97a309b7844aea82b200139febc99998e7f06402 (NixOS) with Nvidia GPU

**Comment by otisdog8** [comment]
Wezterm 979df78 work for me on Hyprland a425fbebe4cf4238e48a42f724ef2208959d66cf with NixOS
Notably the currently released wezterm does not work with that config.

**Comment by Holmgren825** [comment]
Nightly crashes as soon as the window loose focus on river 0.3.5, see #6233. Current stable works as expected.

**Comment by akdor1154** [comment]
Pop 22.04 doesn't crash with default config, but pop's tiling doesn't seem  to work (i'll raise separate issue).

**Comment by cguentherTUChemnitz** [comment]
**tldr:**
So from my side: a new release based on current main branch  would be a great uplift against last release from February, even with known bugs, when running nixos gnome 47 wayland with fractional scaling.
 
long:
I there, i use nixos with gnome 47 in wayland with fractional scaling on a notebook, running with some already known bugs in the bugtracker. To come down to the main question of this thread:

> Thanks for commenting folks. From a triage perspective, what I care about here in the context of this issue is:
> 
>     * _Would releasing nightly now be worse than the prior stable release?_
> 

Yes, i would highly love to see a current release, since i build wezterm at the moment explicitly from github master branch, which would solve for me the recompilation time ;) My current situation between master and latest release from February are:
 - February release does not start in wayland mode with fractional scaling at all, or has completly broken textures for font rendering
 - with current main i get at least a running instance, with some config setup also quite sharp font rendering despite fractional scaling, just the window resizing feels crazy and if have to do it with keyboard shortcuts

 PS: config i use for current main:
 
 ```lua
         -- Your lua code / config here
        return {
          enable_wayland = true,
          font_size = 12.0,
          color_scheme = "Molokai",
          window_decorations = "TITLE | RESIZE",
          font = wezterm.font_with_fallback({
            { family = 'Fira Code' },
            { family = "Cambria Math", scale = 1.0 },
          })
        }
```

**Comment by btstream** [comment]
> I am using Plasma 6 on arch linux, and have found so far two issues:
> 
>     * An already mentioned [Maximize is not working and fullscreen is not fullscreen #6203](https://github.com/wez/wezterm/issues/6203), and it looks a bit messy
> 
>     * Annoying issue with moving wezterm window from one Wayland display to another [Wezterm changes a window's size when moving from display to display #6374](https://github.com/wez/wezterm/issues/6374)

same here

**Comment by g0r3** [comment]
>   [ WezTerm has no icon under Wayland when using fractional scaling (fractional-scale-v1) #6081](https://github.com/wez/wezterm/issues/6081)
>   I don't remember that happening in the last release, but i can not verify that anymore, because the current release is not starting anymore on my system.

I was using [wezterm 20240203.110809.5046fc22-2](https://archlinux.org/packages/extra/x86_64/wezterm/)  from the Arch Linux repository, which corresponds to the latest wezterm release on the Github release page and it doesn't have this issue. This problem appeared for me after compiling the git version of wezterm in order to solve a different problem. So this is very likely a regression. The current git version is still affected.

**Comment by abhinavnatarajan** [comment]
Wezterm 20250103-160645-51886f4c works well for me on Wayland (cosmic-comp and Cosmic DE alpha 5 running on PopOS 24.04). The default configuration works without issue.

**Comment by g0r3** [comment]
> Wezterm 20250103-160645-51886f4c works well for me on Wayland (cosmic-comp and Cosmic DE alpha 5 running on PopOS 24.04). The default configuration works without issue.

Cannot confirm this. I tried it mentioned version on Arch/KDE/Wayland and for me it has the same behavior regarding the icon.

**Comment by Noghpu** [comment]
The nightly version (WezTerm-windows-20250210-212000-f8921727) doesn't launch on my windows machine (10.0.18363 build 18363). Tried installing via scoop and the portable version.

The current release works fine. 

Unfortunately I don't have the rights to update my Windows to see whether a different Windows 10 version would work

**Comment by bew** [comment]
@Noghpu Could you open a dedicated issue for this and fill up the issue template with your environment information?

**Comment by futile** [comment]
Let's just release at this point, and then fix stuff afterwards, then release again (like 1-2-3 weeks later maybe)? It seems like there has been quite some divergence, and that it's difficult to decide what's important enough to make it into this release?

**Comment by wez** [comment]
@futile The reason I haven't made a release yet is because there are a significant number of cases where things are regressions, and I don't want to be under the gun to resolve them because someone was hasty about cutting a release. If you are not making a commitment to actively submit PRs to resolve those issues in that hypothetical 2 week time span, then you don't get to pressure for a release, or make comments about how it isn't moving fast enough for you.

If you are signing up to fix things up in 2 weeks, then I think that's great, but I'd rather spend the 2 weeks now, then cut the release once they are fixed.

**Comment by futile** [comment]
> [@futile](https://github.com/futile) The reason I haven't made a release yet is because there are a significant number of cases where things are regressions, and I don't want to be under the gun to resolve them because someone was hasty about cutting a release. If you are not making a commitment to actively submit PRs to resolve those issues in that hypothetical 2 week time span, then you don't get to pressure for a release, or make comments about how it isn't moving fast enough for you.
> 
> If you are signing up to fix things up in 2 weeks, then I think that's great, but I'd rather spend the 2 weeks now, then cut the release once they are fixed.

Got it, sorry for bothering you, that was unappropriate. Thanks for all your work on Wezterm!

**Comment by Felixoid** [comment]
The latest  version from git crashes #6729, the version `20240203-110809-5046fc22` does not

**Comment by jhonfromerov** [comment]
**Hi!** 
I don't know if this is useful yet, but here is my feedback:
### System details:
- OS: Arch Linux x86_64
- Kernel: Linux 6.13.5-arch1-1
- Compositor: Hyprland 0.47.2-2
- Graphics server: Wayland 1.23.1-1
### Last stable release (Wezterm 20240203.110809.5046fc22-2):
-Wezterm doesn't open from terminal (I try with ghostty and foot) or launcher (wofi 1.4.1-1). 
-You have to apply a workaround like https://github.com/wezterm/wezterm/issues/5604#issuecomment-2190753629.
-When it does open, Starship doesn't work.
### Nightly (https://github.com/wezterm/wezterm/commit/7d0bff0698813c9feeba91a342cd6df94aa34630)
It works fine for me.

**Thank you!**

**Comment by Martin1887** [comment]
#6578 applies here.

**Comment by eatradish** [comment]
Hi, `20240203-110809-5046fc22` is no longer able to reproduce the build, because `wgpu 0.18` was yanked

**Comment by eugenesvk** [comment]
Windows regression: `20250320-072107-a8735851` hangs on receiving a language change message, see https://github.com/wezterm/wezterm/issues/6817

**Comment by klausman** [comment]
Running against this stack (git tags):

drm: libdrm-2.4.125
wayland: 1.24.0
wayland-protocols: 1.45
libliftoff: v0.5.0
libxkbcommon: xkbcommon-1.8.1
wlroots: 0.19.0
labwc: cb79eccea1e259b1979ec4767922256defcda8f3

I get odd maximization bugs, for example if I vertically maximize the terminal, and then focus a different window, the maximization is reverted (goes back to default size). Does not occur with wezterm-nightly.

**Comment by cpence** [comment]
Running on Wayland+labwc here and can say that this nightly fixes a number of resize-related bugs that had been present in the last packaged version for Arch (20240203). Those bugs seem to be related to the constellation of issues described here: https://github.com/labwc/labwc/issues/2087.

AFAICT all of that is remedied on latest nightly. Thanks!

`wezterm -n` also runs fine.

**Comment by danielepintore** [comment]
On the latest update of Hyprland wezterm rendering is broken see #7232 before it was working fine

**Comment by jficz** [comment]
Hyprland 0.51.1
Wezterm 2025-08-14

OS: Debian Trixie and NixOS sharing the same home-manager profile using nixpkgs-unstable.

No (apparent) issues except an occasional "Missing glyph" notification and _sometimes_ a bit slower startup (~1-1.5s).

`wezterm -n` runs fine with a warning:
```
WARN   window::os::wayland::connection > Unable to resolve appearance using xdg-desktop-portal: get_appearance.read_setting: Reading xdg-portal org.freedesktop.appearance color-scheme: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files
```

**Comment by God-damnit-all** [comment]
> Hyprland 0.51.1 Wezterm 2025-08-14
> 
> OS: Debian Trixie and NixOS sharing the same home-manager profile using nixpkgs-unstable.
> 
> No (apparent) issues except an occasional "Missing glyph" notification and _sometimes_ a bit slower startup (~1-1.5s).
> 
> `wezterm -n` runs fine with a warning:
> 
> ```
> WARN   window::os::wayland::connection > Unable to resolve appearance using xdg-desktop-portal: get_appearance.read_setting: Reading xdg-portal org.freedesktop.appearance color-scheme: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files
> ```

So, this means all the release blockers have been resolved, yes?

**Comment by salomon-smekecohen** [comment]
Hello,

For `Verify that the default configuration doesn't crash out.`, are you collecting feedback somewhere besides this issue? Im not sure how useful my report on current nightly would be to you considering the next nightly might invalidate. Are you focusing on `Verify that Wayland support isn't an obvious regression since the last stable release.` instead for now?

<details>
  <summary>My default config executions (OK)</summary>

  ```zsh
  ~ ❯ wezterm -n (OK)
  11:01:59.400  INFO   wezterm_gui::termwindow > QuitApplication over here (window)
  ~ ❯ wezterm -n --config "front_end='WebGpu'" (OK)
  11:02:04.659  INFO   wezterm_gui::termwindow > QuitApplication over here (window)
  ~ ❯ sw_vers
  ProductName:		macOS
  ProductVersion:		26.2
  BuildVersion:		25C56
  ~ ❯ echo $SHELL
  /bin/zsh
  ~ ❯ zsh --version
  zsh 5.9 (arm64-apple-darwin25.0)
  ~ ❯ wezterm -V
  wezterm 20251201-075747-d3b0fdad
```
</details>

Thank you very much.

**Comment by Jarvvski** [comment]
> > Hyprland 0.51.1 Wezterm 2025-08-14
> > OS: Debian Trixie and NixOS sharing the same home-manager profile using nixpkgs-unstable.
> > No (apparent) issues except an occasional "Missing glyph" notification and _sometimes_ a bit slower startup (~1-1.5s).
> > `wezterm -n` runs fine with a warning:
> > ```
> > WARN   window::os::wayland::connection > Unable to resolve appearance using xdg-desktop-portal: get_appearance.read_setting: Reading xdg-portal org.freedesktop.appearance color-scheme: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files
> > ```
> 
> So, this means all the release blockers have been resolved, yes?

Does this mean we can get a release 👀 

it's been a few years

**Comment by shssoichiro** [comment]
A new release would be very nice to have, as there are major issues that have been fixed since the last stable release. Of note, drag-and-drop functionality in KDE Plasma (might be Wayland-specific, not sure) is completely broken in the last stable release, but works in nightly. Please and thanks.

**Comment by manan-gup** [comment]
Information about when a new release might happen is here: https://github.com/wezterm/wezterm/issues/7451#issuecomment-3688958865
In the meantime, you can build and use this PR: https://github.com/wezterm/wezterm/pull/7095
This is a tracking issue, please don't spam.

**Comment by God-damnit-all** [comment]
Note that https://github.com/wezterm/wezterm/issues/7451 was locked by the bot.

#7095 hit a snag where the author states a lack of time to complete the PR:

>> I tried this branch on Ubuntu 24.04 (I know...) and everything _looks_ beautiful but the keyboard input feels quite rough (as if there is a random small delay at times) also when not attempting to repeat a key, just regular typing. Running `wayland-info` I see no mention of `wl_keyboard`. Am I perhaps using a completely different input system somehow? I'm still unfamiliar with Wayland so I don't quite know what I'm looking for.
>
> I have also seen this bug I might have to see where it occurs and fix it. Along with the changes highlighted here in order to complete this PR: https://github.com/wezterm/wezterm/pull/7095#issuecomment-3487726973
>
> With that being said, I am extremely busy with no time left to work on this PR, but if I do get a free day I will come back to complete this PR. The adjustments should be fairly easy. I have no idea when that will be, but if someone wants to pick up on this, feel free.
>
> _Originally posted by @HearseDev in https://github.com/wezterm/wezterm/issues/7095#issuecomment-3737190081_

It's a bit frustrating to see when the project is so close to the finish line for putting out a new stable release.

I'm mentioning it here because people tracking this issue might not be aware the PR is no longer being worked on and that the author of it has offered to let someone else pick up where it was left off on.

**Comment by HearseDev** [comment]
> Note that https://github.com/wezterm/wezterm/issues/7451 was locked by the bot.
> 
> #7095 hit a snag where the author states a lack of time to complete the PR:
> 
> >> I tried this branch on Ubuntu 24.04 (I know...) and everything _looks_ beautiful but the keyboard input feels quite rough (as if there is a random small delay at times) also when not attempting to repeat a key, just regular typing. Running `wayland-info` I see no mention of `wl_keyboard`. Am I perhaps using a completely different input system somehow? I'm still unfamiliar with Wayland so I don't quite know what I'm looking for.
> >
> > I have also seen this bug I might have to see where it occurs and fix it. Along with the changes highlighted here in order to complete this PR: https://github.com/wezterm/wezterm/pull/7095#issuecomment-3487726973
> >
> > With that being said, I am extremely busy with no time left to work on this PR, but if I do get a free day I will come back to complete this PR. The adjustments should be fairly easy. I have no idea when that will be, but if someone wants to pick up on this, feel free.
> >
> > _Originally posted by @HearseDev in https://github.com/wezterm/wezterm/issues/7095#issuecomment-3737190081_
> 
> It's a bit frustrating to see when the project is so close to the finish line for putting out a new stable release.
> 
> I'm mentioning it here because people tracking this issue might not be aware the PR is no longer being worked on and that the author of it has offered to let someone else pick up where it was left off on.

I did not state that it is no longer being worked on. I stated that i would work on it when i have free time to work on it.

**Comment by nakamorichi** [comment]
Why not just release for the platforms that already work? I've been using the nightly build for more than a year on macos and have not experienced noticeable issues. No need to be perfectionist - if want to release perfectly functional app on multiple platforms, such release is probably never going to happen. This is not commercial product any way, so I feel there's no need to stress too much if there are bugs in the releases.

**Comment by God-damnit-all** [comment]
I'm now calling out to the void once more. Could someone continue the #7095 PR? I'm honestly worried about the project's future, stagnating like this makes people lose interest. Anyone?

**Comment by uxfion** [comment]
~~I think this project is already dead, the last stable release was February 4, 2024. Is there a successor? Or should we switch to another project?~~
---
Sorry for not seeing [this issue](https://github.com/wezterm/wezterm/issues/7451#issuecomment-3688958865). I APOLOGIZE. hope wez gets better and get through this 💪. 

But I still think a good open source project should be maintained collectively, rather than being almost stalled because one person is too busy. Some developers should be authorized to approve PRs and set up automatic packaging, given that there are so many Agent tools (claude code, codex) available now.

**Comment by peikk0** [comment]
@uxfion the project is not dead, wez is a bit busy with life stuff at the moment: https://github.com/wezterm/wezterm/issues/7451#issuecomment-3688958865

**Comment by bew** [comment]
> But I still think a good open source project should be maintained collectively, rather than being almost stalled because one person is too busy. Some developers should be authorized to approve PRs and set up automatic packaging, 

I fully agree with this, and as he mentioned in #7451 he's considering setting up something like this (maybe not at much community-based though) but he never had the time to flush out a proposal and designing how it would work exactly at the moment AFAIK..

> given that there are so many Agent tools (claude code, codex) available now.

I don't know Wez's position on this exactly but I'm guessing he's not a fan and would prefer human-based and/or carefully reviewed-by-human contributions. 🤔

---

What's actually worse is the fact that as issues and messages and PRs keep piling up, the backlog of stuff to checkout and sort out on his side are staggering and adds to the mountain of work 😬

**Comment by jficz** [comment]
It's OSS maintained by a single person who has his own problems to solve first, as all of us do. Give Wez a break, or better yet, help him. But not with AI. Not at this point anyway. In my experience, AI slop is hard to review and still isn't advanced enough to actually help the _maintainers_ to keep code quality. Sure, it helps the devs, but I'm sure you've all heard about this little thing called "technical debt" ;)

**Comment by God-damnit-all** [comment]
> In my experience, [AI agents still aren't] advanced enough to actually help the _maintainers_ to keep code quality.

I've noticed that projects which extensively separate frontend and backend code from one another do have luck with using AI agents to review backend code. They're particularly adept at catching edge-cases that humans miss.

They're certainly a lot weaker at reviewing frontend code; so to avoid technical debt, you can bar them from reviewing frontend code completely. But, this is hard to do on many projects that don't go far enough in keeping the frontend and backend code apart. This is probably one of them.

**Comment by bew** [comment]
@God-damnit-all this is not the right place to start discussing stuff nor mention AI-competency, especially because a lot of people are watching this issue about next-release-related information.

👉 Let's keep this issue on topic please everyone, but feel free to open a separate issue for more discussion!

**Comment by adam-ah** [comment]
I appreciate the work that has gone into WezTerm, but the current situation is not sustainable.

The **last release was over two years ago** in February 2024. There are now around **1.5k open issues, 240 open PRs, and 1.4k forks**. Users are sitting on nightlies for basic fixes. Contributors have PRs pending with no clear path forward. Issues and forum discussions keep piling up because there is no visible release process.

Worse, every conversation about this seems to get shut down as the wrong place or wrong time, while the underlying problem remains unresolved. See discussions/issues like #6918, #7451, #7748, and comments here in #6341. If there is no acceptable place to discuss release governance, then that itself is part of the problem.

“Wez is busy” is understandable on a personal level, but it has not, is not, and will not be a valid project strategy. If one person being unavailable can stall the entire project for years, then the governance model is the problem.

The project needs to move forward.

1. Appoint additional maintainers with real merge/release authority.
2. Triage the pending PRs and identify what can be merged, closed, or delegated.
3. Publish the actual release blockers, not a vague moving target.
4. Cut a release candidate from current nightly.
5. Stop blocking every platform on Wayland-specific issues if macOS, Windows are releasable.
6. Delegate CI, packaging, signing, and release chores to trusted contributors.
7. Create a clear public process for contributors instead of leaving PRs and issues to rot.

I need you to understand, this is not about attacking Wez. It is about being honest, the project has outgrown a single-maintainer bottleneck years ago.

If Wez wants to keep maintaining it, great, but still then, he needs to delegate. If he no longer has the capacity, that is fine too, but then new maintainers should be empowered.

WezTerm is too widely used to remain stuck because one person is unavailable. The project needs maintainers, releases, and a path forward now.

If everything stays blocked behind one person, the project will slowly die. New users see a two-year-old release on the homepage or GitHub, assume it is dead, move on, and there is not even a clear link to nightly builds, so all the work sitting in nightlies, PRs, and forks effectively goes to waste.

**Comment by gotgenes** [comment]
@adam-ah I appreciate and share your frustration, but your tone is a bit confrontational and conveys a sense of entitlement. FOSS has always been "Free as in puppy".

Whoever can put in the time and resources to make more frequent releases, and shorten the time to triage Issues and PRs, is going to resolve the situation for WezTerm (or a fork of it). If you can't dedicate that, consider taking a more amiable, empathetic tone.

And, if _nobody_ can put in time and resources to make more frequent releases and manage the issues and PRs, well, the [Red Queen Hypothesis](https://en.wikipedia.org/wiki/Red_Queen_hypothesis) applies to software, too. While WezTerm works today, the project will suffer dependency rot and end up unmaintainable by default. The Big Bang Release is one of the most well-trodden, well-understood software anti-patterns, and it only gets worse the longer the cadence. I think everyone here understands this, on some level. And I think that's where your frustration stems from, too. It's hard to watch a good thing come to an end.

The good news is that we have more choice today than ever. Maybe try Ghostty, or Kitty, or Warp. And we also have access to generative AI, so, maybe fewer people can take on the burden of a project like this; maybe it will be easy for you to generate your own one day.

Do I feel disappointed that my favorite terminal has stagnated? Yes.

Do I feel optimistic the situation will improve or a better alternative will become available? Also, yes!

**Comment by God-damnit-all** [comment]
> conveys a sense of entitlement

To me, it seemed a lot more like a sense of urgency than a sense of entitlement.

**Comment by qiutian00** [comment]
I hope wezterm continues to improve and runs flawlessly.

**Comment by Lebenoa** [comment]
> [@adam-ah](https://github.com/adam-ah) I appreciate and share your frustration, but your tone is a bit confrontational and conveys a sense of entitlement. FOSS has always been "Free as in puppy".
> 
> Whoever can put in the time and resources to make more frequent releases, and shorten the time to triage Issues and PRs, is going to resolve the situation for WezTerm (or a fork of it). If you can't dedicate that, consider taking a more amiable, empathetic tone.
> 
> And, if _nobody_ can put in time and resources to make more frequent releases and manage the issues and PRs, well, the [Red Queen Hypothesis](https://en.wikipedia.org/wiki/Red_Queen_hypothesis) applies to software, too. While WezTerm works today, the project will suffer dependency rot and end up unmaintainable by default. The Big Bang Release is one of the most well-trodden, well-understood software anti-patterns, and it only gets worse the longer the cadence. I think everyone here understands this, on some level. And I think that's where your frustration stems from, too. It's hard to watch a good thing come to an end.
> 
> The good news is that we have more choice today than ever. Maybe try Ghostty, or Kitty, or Warp. And we also have access to generative AI, so, maybe fewer people can take on the burden of a project like this; maybe it will be easy for you to generate your own one day.
> 
> Do I feel disappointed that my favorite terminal has stagnated? Yes.
> 
> Do I feel optimistic the situation will improve or a better alternative will become available? Also, yes!

Seems hostile and disrespect **BUT** I think it's the truth here. I also have a thinking that if you have no interest in maintaining a project anymore, you should transfer ownership or, at least, appoint someone to merge pull requests/do releases, etc. Since a lot of people is still interest in the project, we should be able to recruit someone with the intention of taking over.  

The problem with fork is that it is hard to out-popular official/main package. So the users search for `WezTerm` will eventually landing on this 2 years old release (That's me). And that's why this is more like an urgency. Release without Wayland supports first by hard error like `Wayland is not currently supports` or Supported Wayland but `Expected issues while using with Wayland` or something.

**Comment by dpc** [comment]
Anyone has a time to organize a community fork? Make a Discord where we can gather (yeah, I don't like Discord either, but it is what it is :D ). I'm already running some patched up version of WezTerm and getting frustrated, might as well help contribute.

Just in case this needs to be said: This is not attempting to be hostile. Forking and ability to self serve and have options is a beauty of Open Source. Big respect and thank you for Wez, and I understand if he wants to keep tight ownership of something with his name/nick on it, but personally I would rather be using something with more self-serve and fast paced dev process. Lets gather, count how many people are willing to help and do something. ATM I don't want to be a leader in there, but over time I might consider if no one with more available time steps up.

The two forks can freely copy from each other so it might benefit everyone, even if they diverge and keep being developed at their own pace.

**Comment by bew** [comment]
> [!WARNING]
> #### (edit @2026-06-04)
> # This message is now outdated!
> See Wez' message below https://github.com/wezterm/wezterm/issues/6341#issuecomment-4620396384


-----

I've mentioned it in pr somewhere IIRC, but I've actually been thinking seriously about spinning up a community fork, realistically it might be a few weeks before anything concrete happens (I'm actually moving home this month ^^'), but it's moved from "maybe someday" to "probably yes" in my head. 🙃

.. Wez has been unreachable across all communication channels around this project for a while now, waiting doesn't feel viable anymore.. :/
If Wez comes back and we managed to get something going we might revive this project!

**When I do move forward**, my immediate priorities would be pretty pragmatic I think:
Get the ~30-40 open PRs that have been sitting around actually merged, set up a basic community space, figure out a release process, and getting a base contribution flow.

**Longer term** I have a rough vision:
Test coverage, more guides in the docs, domain model rework (per-pane anyone?), config/Lua API improvements, trimming high-cost low-usage features, reduce custom code by switching to popular/maintained crates (e.g. for the UI layers).

👉 I'll open a separate issue to discuss this more concretely when I'm ready — if you're interested in contributing (code, triage, docs, anything really), that'd be where to drop a comment. 🙂

👉 Try to avoid adding replies in this thread in the meantime 🙏
Add reactions to this message if you want!

_(On Discord: I'm not a fan either, but I'll leave the "where we gather" discussion for later once there's actually something to gather around)_

**Comment by Destroy666x** [comment]
> Seems hostile and disrespect

There is nothing hostile about it... It's just objectively listing facts that many software authors on GitHub are forgetting about. They could be summed up with one word: "responsibility". Tons of open source devs lack it. Passing projects to active and trusted contributors should be promoted wherever possible, so that projects don't end up with tons of security, vulnerabilities, die or get sold to shady companies in the end. People need to get out of their bubbles and accept that there's something between praising/staying silent and disrespect/hate/whatever - it's called "constructive criticism".

**Comment by wez** [comment]
I just wanted to chime in here and say that I'm back after a bit of a hiatus.
It's been a challenging period for me and my family this past couple of years. We've faced multiple bereavements, a move between countries, a change in employment status, and just general "WTF" with the state of the world and all of the associated overheads with all of that.

I want to thank @bew for keeping the lights on to the limited extent that the permissions here enabled them to act while I was unavailable and unresponsive.

I haven't read all the comments in this thread, but I did see someone mention something about "tightly controlling" this project.  I wanted to clarify that I'm not so much controlling as I am feeling very responsible.   This sort of software is on what I think of as "the sharp end" of systems programming: much of it is in low-level and highly technical areas that are not broadly well understood and because it is cross platform, there is a lot to consider when reviewing issues and PRs.   This makes it more time consuming to review and harder to move with as much confidence as you might in projects that work in different levels of the technical stack.  Given my limited time, I didn't want to accept a PR that would cause a regression and make things worse off than they were... and not have the time to revisit and resolve it.

What we're doing about this:

* I've upgraded @bew's access so that they are empowered to make changes here without blocking on me and without the need to fork to make things happen.
* We're working up a strategy to unblock the review and merging of PRs, part of which will involve considering how to scale our very small core team

More tactically, this particular giant issue has outlived its usefulness, so we'll be tracking various next steps in a combination of issues and/or github project tracking; we'll update here as we make progress on that.

**Comment by perapp** [comment]
So happy to see you here again @wez! Brings me new hope my all time favorite terminal will live on in the future and I might not need to look elsewhere after all ❤️

**Comment by atahrijouti** [comment]
@wez I hope things are heading towards a more positive outlook :-)
As you said, a terminal isn't one's average piece of software, we trust you 👍🏼

**Comment by swarnimarun** [comment]
I just caught up on the recent messages. I’m not sure whether I was the one who said “tightly controlled,” either in this thread or another one, but I definitely remember thinking it.

I understand the criticality of the project, believe me I realize the issues, I use wezterm on everything all my systems, I use it now more than ever, since the rise of vibe coding and terminal agents, not like I used it any less before given I am/was/will be a vim/helix text editor user.

To cut it short: I’d genuinely like to help however I can to keep the project maintained. I currently fix minor bugs in my own fork when I have time, though I haven’t been pushing those changes upstream because I do use AI and my understanding of the project is definitely nowhere close to perfect. But I can at least document the bugs and my solutions, then share them in the relevant issue threads when time allows.

---

I’d also be happy to throw in some money toward bugs I’d like to see fixed properly, whether through bounties or some other arrangement.

I understand that development may move more slowly than some people would like. Personally, I’m less interested in big new features than in bug fixes, stability, security updates, and improving the plugin and configuration experience.

**If there is anything I, and hopefully others here, can do to help,** a blog post outlining what would be useful would be greatly appreciated. Please link it here if you write one.

As long as I can keep using WezTerm reliably across my systems and receive bug fixes and security updates regularly, I’m more than happy for major features to be deprioritized. The rest, we can plugin our way to success.


**Aside:**

Thank you for building and maintaining such an amazing project. I know maintaining something this important to so many people is no small thing, and I’m grateful for all the work that has gone into it. Wishing everyone involved the very best, and I hope you all get the time, support, and space needed to keep the project healthy.

**Comment by rvsfirecreeper** [comment]
Gotta say, the nightly is 1000% better on Tiling Wayland
Before it would break because of WezTerm expecting a 0-size signal
On Tiling now it works properly

**Comment by manan-gup** [comment]
Hi, @wez & @bew, can this thread be restricted to contributors and general comments be moved to discussions? The comments here generate a lot of unnecessary notifications for people who are subscribed to this thread for development updates. Thanks for your efforts in restarting maintenance for wezterm!
